### DISCUSSION TITLE
difference between time64 columns

### DISCUSSION THREAD

**Discussion by lmullany** [h2oai/datatable#3218] [Q&A] [answered]
What is the best/fastest way to add a duration column that indicates the difference between two time64 columns?
```
import datatable as dt
import numpy as np
from datetime import datetime as d
DT = dt.Frame(t1 = [d(2020, 5, 11, 12, 0, 0)], t2 = [d(2020, 5, 14, 16, 10, 59)])
```

To make a new column that represents the minutes between t1 and t2, do I need to do something like this?
```
DT["dur"] = (DT["t2"].to_numpy() - DT["t1"].to_numpy())/np.timedelta64(1,'m')
```

Is there a better way?

**Comment by oleksiyskononenko** [comment, marked as answer]
At some point we will add a new `duration` type, so that one could directly calculate difference between the time columns, or between the date columns. 

However, what you're asking is possible to be done in datatable even now. That's because internally `time64` column is stored as `int64`. So you can do something like that to calculate the number of minutes

```
DT["duration"] = DT[:, (dt.int64(dt.f.t2) - dt.int64(dt.f.t1))/(60*10**9)]
```

**Reply by lmullany** [reply]
Thanks much!
